easybuild/tools/module_generator.py

```diff
@@ -422,28 +430,38 @@ def det_installdir(self, modfile):
 
         return res
 
-    def unpack_setenv_value(self, env_var_name, env_var_val):
+    def unpack_setenv_value(self, *args, **kwargs):
+        """
+        DEPRECATED method, should not be used.
+        Replaced with (internal) _unpack_setenv_value method.
+        """
+        self.log.deprecated("unpack_setenv_value should not be used directly (replaced by internal method)", '6.0')
+        value, use_pushenv, _ = self._unpack_setenv_value(*args, **kwargs)
+        return value, use_pushenv
+
+    def _unpack_setenv_value(self, env_var_name, env_var_val):
         """
         Unpack value that specifies how to define an environment variable with specified name.
         """
-        use_pushenv = False
+        use_pushenv = self.DEFAULT_MODEXTRAVARS_PUSHENV
+        resolve_shell_vars = self.DEFAULT_MODEXTRAVARS_SHELL_VARS
```

Comment:
@lexming After another pre-merge review, I realized we could play our naming game better here, see https://github.com/lexming/easybuild-framework/pull/10